src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -665,13 +660,18 @@ final case class ProcessApiServiceImpl(
   private def convertToFlattenEservice(
     eservice: client.model.EService,
     agreements: Seq[AgreementManagementDependency.Agreement],
-    description: String
+    tenant: Future[TenantManagementDependency.Tenant]
   ): Seq[FlatEService] = {
     val flatEServiceZero: FlatEService = FlatEService(
       id = eservice.id,
       producerId = eservice.producerId,
       name = eservice.name,
-      producerName = description,
+      producerName = tenant
+        .onComplete({
+          case Success(value) => value.name
+          case Failure(_)     => ""
+        })
+        .toString,
```

Comment:
Okay, this part is to rewrite. You should never depend on a `Future` in a signature, you might not know whether it's completed, you might need to wait for its completion and it's not a caller's duty to decide when and where to wait for a future completion. It's the same logic behind the `never pass an Option as an argument.

I suggest you to rewrite `tenant: Future[TenantManagementDependency.Tenant]` as `tenant: TenantManagementDependency.Tenant` and to flatmap `tenantManagementService.getTenant(service.producerId)` where you call this function.

For what concerns the content, instead of using that pattern match, you may write `tenant.map(_.value).getOrElse("")` but to do that again in the calling function and not here.

Ping me it something it's not clear :)